app/mirage/config.js

```diff
@@ -298,6 +298,20 @@ export default function() {
       return new Mirage.Response(400, {}, {errors: errors});
     });
 
+    this.post('/upload', function(db, request) {
```

Comment:
This is where your test are failing.  Signature should be 
```javascript
this.post('/upload', function() {
```
Since request and db arguments are not used in the function.